PR/redux login

alonss0 opened the pull request:
Check commit description for detail information

smarquezmgh committed:
the mgh dao text changed by the hub dao

nanSer0 committed:
Merge pull request #219 from metagamehub/PR/the-hub-dao-text

the mgh dao text changed by the hub dao

alonss0 committed:
Login rebuild

Some changes has been added in order to make clean Login feature.

Login component manage login state and renders connect button from rainbowkit.

Store init for state management has been added.

Layout file should include `use client` as we need to use client react hooks on child components.

Left to check:
 - ITRM auth web3 service for access token generation. This includes a backend hook named `useToken` from `old/backend` files.

alonss0 committed:
adding arcana connector

alonss0 committed:
Fixing arcana connector

This fix Arcana Connector TypeError as suggested from Arcana Members we should use the next version of auth-wagmi package "@arcana/auth-wagmi": "^2.0.0-beta.1" this is included in package.json

smarquezmgh committed:
no functional parts hidden

smarquezmgh committed:
added dependencie deleted

alonss0 committed:
Updating file locations

nanSer0 committed:
Merge pull request #228 from metagamehub/PR/hide-no-functional-parts

no functional parts hidden

nanSer0 commented:
@Alonsso add "s" at the end of "component" folder name

alonss0 committed:
Login backend and watchlist updates

Migration of old backend code review.
We were using web3auth to get an access token through a server. This creates conflicts when trying to migrate big part from Watchlist and Portfolio slices. The access token is needed as param in order to make data requests. Also, auth service is managing different routes for watchlist and portfolio GET's.

With new packages versions, things has changed. e.g.: useSigner from Wagmi is deprecated and all Signer types should be fixed to use new type called WalletClient.

The above, creates a big snowball. And in order to make code clean and use recent package updates. I purpose using Rainbowkit authentication for session management and verify AuthService Watchilist and Portfolio routes as those should manage another type of data.

OswaldUnity committed:
Update pixi version package json fix error on build

alonss0 committed:
Minor changes fixes

OswaldUnity committed:
Revert "Update pixi version package json fix error on build"

This reverts commit d3f125a09130d3e2ff59463b2b7dda83fb076c9c.

OswaldUnity committed:
Removed the only one that actually breaks everything

DAC-info committed:
Merge pull request #230 from metagamehub/PR/update_package_json_fix_error_pixi

Update package json fix error pixi

alonss0 committed:
adding watchlist fetch

The access token is needed to get watchlist from auth service, I tried in previous commits the token generation, but for some reason, we are unable to get the access token, as new type WalletClient inside web3auth service is not finding the getAdresses method.

Still looking a for a way to make the fetch without needing the access token or getting it.

alonss0 committed:
Obtaining access token

Clean up code from web3 auth service, new implementation of Wallet Client to sign message and get access token.

To fix: Arcana Auth is not creating a sign message and we are not able to get access token in this case.

alonss0 committed:
portfolio migration

Adding needed files to make the portfolio fetch.

Left to check a way to make it clean.

alonss0 committed:
Merge branch 'dev' into PR/redux-login

alonss0 committed:
erasing comments

alonss0 committed:
Revert "Merge branch 'dev' into PR/redux-login"

This reverts commit 9e4ed7eee5a98f1b7577ced594ea0370d6297b2a, reversing
changes made to 71f89ac94c787231bb06258276c3c78efa690eb5.

alonss0 committed:
Merge branch 'refactoring' into PR/redux-login

alonss0 committed:
Merge branch 'PR/redux-login' of https://github.com/metagamehub/MGH-APP-V2.0 into PR/redux-login

nanSer0 reviewed (dismissed):
Make changes based on comments on the meet

alonss0 committed:
Fixing types

In this large commit, some types has been fixed, however, left to check some build errors, due to any json type response and Portfolio refactoring.

Viem and wagmi packages updated as useWalletClient hook was not working I decided to use getWalletClient from wagmi/core.

Unused files and code removed.

alonss0 committed:
Portfolio slice types fix

alonss0 committed:
Cleaning up unused code and error fixes

alonss0 committed:
More fixes

alonss0 committed:
Updating logs and build errors

alonss0 committed:
Merge branch 'refactoring' into PR/redux-login

alonss0 committed:
Some syntax error fixes

OswaldUnity commented on package.json:
```diff
@@ -10,21 +10,23 @@
   },
   "dependencies": {
+    "@arcana/auth": "^1.0.5",
+    "@arcana/auth-wagmi": "^2.0.0-beta.1",
```
Please move the packages that you know you are using at the start of the dependencies array, that way we know which ones we need.

OswaldUnity commented on package.json:
```diff
@@ -10,21 +10,23 @@
     "@pushprotocol/restapi": "^1.3.8",
-    "@rainbow-me/rainbowkit": "^1.0.2",
+    "@rainbow-me/rainbowkit": "^1.0.4",
+    "@rainbow-me/rainbowkit-siwe-next-auth": "^0.2.0",
```
If you are not using this packages please remove them.

OswaldUnity commented on src/backend/services/AuthService.ts:
```diff
@@ -0,0 +1,58 @@
+import { LogError } from '../../utils/logging.util';
+
+interface nonceResponse {
+    nonce: number
```
interfaces have semicolons at the end of every property

OswaldUnity commented on src/backend/services/AuthService.ts:
```diff
@@ -0,0 +1,58 @@
+import { Module } from '../../enums/logging.enum';
+import { LogError } from '../../utils/logging.util';
+
+interface nonceResponse {
```
Interface names follow PascalCase

OswaldUnity commented on src/backend/services/AuthService.ts:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    fetchNonce = async (address: string) => {
+        const nonceRes = await fetch(
```
Since we are using axios, try to avoid fetch

OswaldUnity commented on src/backend/services/ITRMService.ts:
```diff
@@ -0,0 +1,58 @@
+}
+
+interface CoinValues {
+  [key: string]: CoinValue;
```
this type is called a Record

OswaldUnity commented on src/backend/services/AuthService.ts:
```diff
@@ -0,0 +1,58 @@
+interface nonceResponse {
+    nonce: number
+}
+class AuthService {
```
Try to avoid classes and move the different methods to functions

OswaldUnity commented on src/backend/services/ITRMService.ts:
```diff
@@ -0,0 +1,58 @@
+
+export async function getCoingeckoPrices(): Promise<ICoinPrices> {
+  let coinValues: CoinValues = {
+    "decentraland": {
```
move the keys to an enum

OswaldUnity commented on src/backend/useToken.ts:
```diff
@@ -0,0 +1,82 @@
+
+const REQ_URL = `${process.env.AUTH_SERVICE ?? ""}/authService/`;
+
+export const axiosBase = axios.create({
```
Remove export when no one outside of this file is using it

OswaldUnity commented on src/component/login.component..tsx:
```diff
@@ -0,0 +1,98 @@
+"use client"
+
+import React, { useState, useEffect, useCallback } from 'react'
```
avoid importing React from 'react'

OswaldUnity commented on src/component/login.component..tsx:
```diff
@@ -0,0 +1,98 @@
+
+import { useAppDispatch, useAppSelector } from '../state/hooks'
+
+import * as loginActions from '../state/loginSlice'
```
Avoid * on imports

OswaldUnity commented on src/component/login.component..tsx:
```diff
@@ -0,0 +1,98 @@
+import { fetchPortfolio } from '../state/portfolioSlice';
+import { WalletClient } from 'viem';
+
+let didSignerSet = false;
```
component level variables start with "_"

OswaldUnity commented on src/component/login.component..tsx:
```diff
@@ -0,0 +1,98 @@
+
+let didSignerSet = false;
+
+export function Login() {
```
"export default" for components

OswaldUnity commented on src/backend/useTokenExpiration.ts:
Follow file name nomenclature, files should have functionality (goal) name plus "." plus type of file
Example:
Component Hello
Filename should be "hello.component.ext"

OswaldUnity commented on src/backend/useToken.ts:
Follow file name nomenclature, files should have functionality (goal) name plus "." plus type of file
Example:
Component Hello
Filename should be "hello.component.ext"

OswaldUnity commented on src/backend/services/ITRMService.ts:
This services won't execute on backend

OswaldUnity commented on src/enums/logging.enum.ts:
```diff
@@ -2,4 +2,9 @@
   ValuationColoring = 'ValuationColoringUtil',
   LandSocket = 'LandSocketUtil',
+  ArcanaProvider = 'ArcanaProviderUtil',
+  PortfolioSlice = 'PortfolioSliceFetch',
```
Use values that actually describe the module, in this case this is a redux slice (I don't understand the fetch use here)

OswaldUnity commented on src/enums/logging.enum.ts:
```diff
@@ -2,4 +2,9 @@
   LandSocket = 'LandSocketUtil',
+  ArcanaProvider = 'ArcanaProviderUtil',
+  PortfolioSlice = 'PortfolioSliceFetch',
+  WagmiConfig = 'WagmiConfig',
```
This is wrong as well, doesn't represent where the error is happening

OswaldUnity commented on src/interfaces/common.interface.ts:
```diff
@@ -0,0 +1,11 @@
+}
+
+export interface WatchlistResponse {
+    "decentraland": object,
```
Avoid the usage of object please

OswaldUnity commented on src/lib/chains.ts:
```diff
@@ -0,0 +1,151 @@
+import { IChainData } from './types'
+
+export const Chains = {
```
Needs to have a type please

OswaldUnity commented on src/lib/contracts.ts:
```diff
@@ -0,0 +1,43 @@
+
+import AxieLandAbi from '../backend/abi/AxieLand.json'
+
+export const Contracts = {
```
Needs to have a type please

OswaldUnity commented on src/lib/enums.ts:
```diff
@@ -0,0 +1,31 @@
+export enum Network {
```
Enum properties PascalCase please

OswaldUnity commented on src/lib/enums.ts:
Move or rename file with a more related name

OswaldUnity commented on src/lib/nftUtils.ts:
```diff
@@ -0,0 +1,87 @@
+) => {
+  if (!address?.startsWith('0x')) return;
+
+  let contracts: {
```
Needs to have a type please

OswaldUnity commented on src/lib/nftUtils.ts:
```diff
@@ -0,0 +1,87 @@
+ * */
+export const getUserNFTs = async (
+  provider: Provider,
+  providerChainName: 'Ethereum' | 'Polygon' = 'Ethereum',
```
Create type apart (preferably an enum)

OswaldUnity commented on src/lib/types.ts:
```diff
@@ -0,0 +1,49 @@
+import { Metaverse } from './metaverse'
+
+export interface IAssetData {
+  name: string
```
Interface properties end with a semicolon

OswaldUnity commented on src/lib/types.ts:
Interfaces should be on an interface folder

OswaldUnity commented on src/lib/utilities.ts:
```diff
@@ -0,0 +1,8 @@
+/**
+ * @returns Array of Object keys with their proper types. Use this instead of Object.keys
+ */
+export function typedKeys<O extends object, K extends keyof O = keyof O>(
```
Avoid at all costs

OswaldUnity reviewed (changes requested):
Lots of things to do, sadly not much time
